crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1411,30 +1411,54 @@ impl Client {
         }
     }
 
+    /// Finish joining a room.
+    ///
+    /// If the room was an invite that should be marked as a DM, will include it
+    /// in the DM event after creating the joined room.
+    async fn finish_join_room(&self, room_id: &RoomId) -> Result<Room> {
+        let mark_as_dm = if let Some(room) = self.get_room(room_id) {
+            room.state() == RoomState::Invited
```

Comment:
won't this race with the `/sync` response, causing it to do the wrong thing if the `/sync` response arrives before the `/join` completes?